code/code/spec/spec_objs.cc

```diff
@@ -5729,6 +5729,29 @@ int regeneration(TBeing *ch, cmdTypeT cmd, const char *, TObj *o, TObj *)
   return FALSE;
 }
 
+int pietyRegen(TBeing *ch, cmdTypeT cmd, const char *, TObj *o, TObj *)
+{
+  if (!o)
+    return FALSE;
+  if (!(ch = dynamic_cast<TBeing *>(o->equippedBy)))
+    return FALSE;       // item not equipped (carried or on ground)
+
+  if(cmd == CMD_GENERIC_PULSE) {
+
+    // don't spam if we're already full
+    if (ch->getPiety() >= ch->pietyLimit())
```

Comment:
Not sure I understand this.

The ch = (dynamic_cast<TBeing *>(o->equippedBy)) part?

Reply:
Eh, I guess it doesn't matter much.

Reply:
Yeah I thought that was strange too but I copied it from elsewhere. I can't think how it would break things but if there's a safer way to test that the object is equipped without overwriting the ch that was passed in I'm happy to change this and update anywhere else it may be used.